server/controllers/userAuth.js

```diff
@@ -465,6 +465,24 @@ const checkToken = (req, resp) => {
     .catch(() => resp.sendStatus(400));
 };
 
+const checkIfDataLeft = async (req, resp) => {
+  const { _id: userId } = req.user;
+
+  try {
+    const errors = await checkIfUserIsTheOnlyOwner(List, Cohort, userId);
```

Comment:
maybe this `checkIfUserIsTheOnlyOwner` is no longer needed? Before, deleteAccount controller was very complicated and log so I extract all its parts to separate helpers. Now. this is separate controller so all its logic can be written here

Reply:
Let leave it as it is. Another layer of polymorphism is expected, especially for these kinds of checks. We never know if we gonna check this information in some other places. Even if not, it will be much easier to test and reason about.